lib/aiko/led.py

```diff
@@ -71,8 +72,35 @@ def apply_dim(color, dimmer=None):
 # TODO: Now fails when called by exec(), see lib/aiko/mqtt.py
 # return tuple([int(element * dimmer) for element in color])
 
-def fill(color):
+# Allow setting dim from code or MQTT
+def set_dim(dimmer):
+    global dim
+    dim = dimmer
+
+# this is used to turn the neopixels back on to default brightness
+def reset_dim():
+  parameter = configuration.main.parameter
+  set_dim(parameter("dim", configuration.led.settings))
+
+# Take from -0.9 to 0.9 (+-0.1 is more typical) and adjust diming
+# value. Make sure it stays within 0 to 1
+def change_dim(change):
+    global dim
+    dim += change
+    if dim<0: dim=0
+    if dim>1: dim=1
+
+def print_dim():
+    print("Dim: ", dim)
+
```

Comment:
I think they can figure this out themselves?
```suggestion
```

Reply:
after the fact, it looks a bit stupid, but I added this:
- while dealing with the variable having different values depending on where it was used
- my more general habit that accessing other objects' variables should be through methods, not direct access.

If Andy doesn't like this, I can remove it.

Reply:
I'm generally fine skipping the encapsulation, but if we're doing it, strikes me we could propagate the OLED proxy style? It uses a single function or method as both getter and setter by taking an argument with a default of None, setting the value only if you give one, and returning it either way. Does that work for you, too?

Reply:
sounds reasonable. honestly, I'll go with whatever make Andy happy since that's his branch :)